lib/neovim-plugin.nix

```diff
@@ -95,17 +95,19 @@ rec {
           mkRenamedOptionModule (basePluginPath ++ optionPath) (settingsPath ++ optionPathSnakeCase)
         ) optionsRenamedToSettings);
 
-      options.${namespace}.${name} = {
-        enable = mkEnableOption originalName;
-
-        package = nixvimOptions.mkPluginPackageOption originalName defaultPackage;
-
-        settings = mkSettingsOption {
-          pluginName = name;
-          options = settingsOptions;
-          example = settingsExample;
-        };
-      } // extraOptions;
+      options.${namespace}.${name} =
+        {
+          enable = mkEnableOption originalName;
+          package = nixvimOptions.mkPluginPackageOption originalName defaultPackage;
+        }
+        // optionalAttrs (settingsOptions != null) {
```

Comment:
We could instead have a `hasSettings ? true` argument, I just figured you'd prefer making the existing arg nullable.

Reply:
I don't know. Maybe the `hasSettings` will be less prone to unwanted behavior.

Reply:
Done